sdk/core/Azure.Core.TestFramework/src/RecordMatcher.cs

```diff
@@ -171,11 +183,24 @@ private int CompareBodies(byte[] requestBody, byte[] responseBody, StringBuilder
             IsEquivalentUri(entry.RequestUri, otherEntry.RequestUri) &&
             CompareHeaderDictionaries(entry.Request.Headers, otherEntry.Request.Headers, VolatileHeaders) == 0;
 
-        private static bool AreUrisSame(string entryUri, string otherEntryUri) =>
-            // Some versions of .NET behave differently when calling new Uri("...")
-            // so we'll normalize the recordings (which may have been against
-            // a different .NET version) to be safe
-            new Uri(entryUri).ToString() == new Uri(otherEntryUri).ToString();
+        private bool AreUrisSame(string entryUri, string otherEntryUri) =>
+            TranslateUri(entryUri) == TranslateUri(otherEntryUri);
+
+        private string TranslateUri(string uriToTranslate)
+        {
+            var req = new RequestUriBuilder();
+            var uri = new Uri(uriToTranslate);
+            req.Reset(uri);
+            req.Query = "";
+            NameValueCollection queryParams = HttpUtility.ParseQueryString(uri.Query);
+            foreach (string param in queryParams)
+            {
+                req.AppendQuery(
+                    param,
+                    VolatileQueryParameters.Contains(param) ? VolatileValue : queryParams[param]);
+            }
```

Comment:
Considered having a separate path for the 99% case where there are no VolatileQueryParameters but I figured "why not just commit?"